base/libgit2/tag.jl

```diff
@@ -35,6 +35,7 @@ function name(tag::GitTag)
 end
 
 function target(tag::GitTag)
-    oid_ptr = Ref(ccall((:git_tag_target_id, :libgit2), Ptr{Oid}, (Ptr{Void}, ), tag.ptr))
-    return oid_ptr[]
+    oid_ptr = ccall((:git_tag_target_id, :libgit2), Ptr{Oid}, (Ptr{Void}, ), tag.ptr)
+    oid_ptr == C_NULL && return Oid()
```

Comment:
this shouldn't happen at all, `git_tag_target_id` will always return pointer to tag's tagret object id, unless something very wrong would happen during the tag parsing and no error was triggered (highly unlikely).